src/components/AlbumItem.jsx

```diff
@@ -35,6 +35,9 @@ export class AlbumItem extends Component {
         url: link,
         isLoading: false
       }))
+      .catch(linkError => {
```

Comment:
I thank that `fetchPhoto` was returning a Promise but it is not the case. At this point, it is almost just a wrapper for `getPhotoLink`.  As I think that the real purpose of this method is to fetch the main photo of the album, what do you think about put the related logic inside it, like this:

```javascript
fetchMainPhoto(album) {
  if (album.photoIds && album.photosIds.length) {
       return getPhotoLink(album.photoIds[0])
  } else {
       return Promise.resolve(null)
  }
}
```
And call it like this :

```javascript
fetchMainPhoto(props.album || {})
  .then(link => this.setState({
    url: link,
    isLoading: false
  })
  .catch(linkError => {
    this.props.onServerError(linkError)
    this.setState({
      url: '',
      isLoading: false
    })
  })
```
Bonus : you don't need to bind this method.